molecule/verifier/lint/precommit.py

```diff
@@ -0,0 +1,134 @@
+#  DEALINGS IN THE SOFTWARE.
+import os
+import sh
+from molecule import logger
+from molecule import util
+from molecule.verifier.lint import base
+
+LOG = logger.get_logger(__name__)
+
+
+class PreCommit(base.Base):
+    """
+    Pre-commit linter class to execute pre-commit tool for linting files.
+
+    `PreCommit`_ is not the default verifier linter.
+
+    `PreCommit`_ is a linter for python files and more.
+
+    Additional options can be passed to ``pre-commit`` through the options
+    dict.  Any option set in this section will override the defaults.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: testinfra
+          lint:
+            name: pre-commit
+            options:
+              remove-tabs:
+
+    Test file linting can be disabled by setting ``enabled`` to False.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: testinfra
+          lint:
+            name: pre-commit
+            enabled: False
+
+    Environment variables can be passed to lint.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: testinfra
+          lint:
+            name: pre-commit
+            env:
+              FOO: bar
+
+    .. _`Pre-Commit`: https://pre-commit.com/
+    """
+
+    def __init__(self, config):
+        """
+        Set up the requirements to execute `pre-commit` tool.
+
+        :param config: An instance of a Molecule config.
+        """
+        super(PreCommit, self).__init__(config)
+        self._precommit_command = None
+        if config:
+            self._tests = self._get_tests()
+
+    @property
+    def default_options(self):
+        return {}
+
+    @property
+    def default_env(self):
+        return util.merge_dicts(os.environ.copy(), self._config.env)
+
+    def bake(self):
+        """Bake a `pre-commit` command so it's ready to execute."""
+        self._precommit_command = sh.Command('pre-commit').bake(
+            'run',
+            self.options,
+            '--files',
+            self._tests,
+            _env=self.env,
+            _out=LOG.out,
+            _err=LOG.error)
+
+    def execute(self):
+        if not self.enabled:
+            msg = 'Skipping, verifier_lint is disabled.'
+            LOG.warn(msg)
+            return
+
+        if not self._tests:
+            msg = 'Skipping, no tests found.'
+            LOG.warn(msg)
+            return
+
+        if self._precommit_command is None:
+            self.bake()
+
+        msg = 'Executing pre-commit on files found in {}/...'.format(
+            self._config.verifier.directory)
+        LOG.info(msg)
+
+        try:
+            util.run_command(self._precommit_command, debug=self._config.debug)
+            msg = 'Lint completed successfully.'
+            LOG.success(msg)
+        except sh.ErrorReturnCode as e:
+            util.sysexit(e.exit_code)
+
+    def _get_tests(self):
+        """
+        Walk the verifier's directory for tests and returns a list.
```

Comment:
Docstring title shouldn't attempt to explain implementation details. It should mention the intention/high-level effect, nothing more. Everything else should go into a separate description paragraph.